### PR TITLE
Client changes

### PR DESCRIPTION
**Pull request by MohamedShatry**
Changes may also reflect changes on the server branch since this branch is dependent on the server branch

### CODE REVIEW

File: portfolio/src/main/webapp/comments-script.js [hunk 1]
```diff
@@ -2,6 +2,79 @@
 function callFetch(){
     fetch("/data")
     .then(res => res.json())
-    .then(res => console.log(res))
+    .then(res => {
+        console.log(res);
```
Comment (addarsh-google) [inline portfolio/src/main/webapp/comments-script.js]: This is probably not needed anymore.

File: portfolio/src/main/webapp/comments-script.js [hunk 2]
```diff
@@ -2,6 +2,79 @@
...
+    fetch("/data", {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(data)
+    })
+    .then(res => res.json())
+    .then(res => {
+        console.log("Got response");
```
Comment (addarsh-google) [inline portfolio/src/main/webapp/comments-script.js]: Log statements probably not needed anymore.

File: portfolio/src/main/webapp/comments-script.js (hunk 1 above)
Comment (MohamedShatry) [reply portfolio/src/main/webapp/comments-script.js]: Ooops! My bad. Will fix that. Resolved

File: portfolio/src/main/webapp/comments-script.js (hunk 2 above)
Comment (MohamedShatry) [reply portfolio/src/main/webapp/comments-script.js]: True. Resolved

File: portfolio/src/main/webapp/comments-script.js [hunk 3]
```diff
@@ -45,7 +45,7 @@ function createCommentElement(comment){
 
     commentContainer.appendChild(commentTag);
     commentContainer.appendChild(bottom_div);
-
+
```
Comment (addarsh-google) [inline portfolio/src/main/webapp/comments-script.js:73]: Is this commit as intended?

File: portfolio/src/main/webapp/comments-script.js [hunk 4]
```diff
@@ -1,7 +1,121 @@
...
+    error_tag.classList.add("error-tag");
+    const err_text = document.createTextNode("Ooops! Looks like there are no comments to show! Type one");
+    error_tag.appendChild(err_text);
+
+    main.appendChild(error);
+    main.appendChild(error_tag);
+}
+
+//This function converts the unix timestamp to date and time format
+function getDateTimeFromTimestamp(unixTimeStamp) {
```
Comment (addarsh-google) [inline portfolio/src/main/webapp/comments-script.js]: is there a library that can do this?

File: portfolio/src/main/webapp/comments-script.js [hunk 5]
```diff
@@ -1,7 +1,121 @@
...
+    var date = new Date(unixTimeStamp);
+    return ('0' + date.getDate()).slice(-2) + '/' + ('0' + (date.getMonth() + 1)).slice(-2) + '/' + date.getFullYear() + ' ' + ('0' + date.getHours()).slice(-2) + ':' + ('0' + date.getMinutes()).slice(-2);
+}
+
+//Add an event listener to the submit button that send the response as JSON to the server
+//Later, this will allow us to add data from cookies
+
```
Comment (addarsh-google) [inline portfolio/src/main/webapp/comments-script.js:103]: nit: space between description and callback.

File: portfolio/src/main/webapp/comments-script.js (hunk 3 above)
Comment (MohamedShatry) [reply portfolio/src/main/webapp/comments-script.js:73]: Yep. Everything looks intended to me. I'm sorry if I've missed the error

File: portfolio/src/main/webapp/comments-script.js (hunk 4 above)
Comment (MohamedShatry) [reply portfolio/src/main/webapp/comments-script.js]: I think momentjs can do that. Let me look into importing it

File: portfolio/src/main/webapp/comments-script.js (hunk 3 above)
Comment (MohamedShatry) [reply portfolio/src/main/webapp/comments-script.js:73]: It looks like what I intended it to be. I'm sorry f I'm missing something

File: portfolio/src/main/webapp/comments-script.js (hunk 5 above)
Comment (MohamedShatry) [reply portfolio/src/main/webapp/comments-script.js:103]: Will fix. Resolved

File: portfolio/src/main/webapp/comments-script.js (hunk 4 above)
Comment (octopus1000) [reply portfolio/src/main/webapp/comments-script.js]: > This function converts the unix timestamp to date and time **format**

Can you also comment on what specific format you are looking at?
momentjs can be a bit heavy if we are not playing with dates a lot.

Consider using supported format. date.toISOString() if acceptable.

If certain format must be followed, suggest having a test for this function.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -22,27 +28,72 @@
...
+            CommentClass commentInstance = new CommentClass(comment, userName, timestamp);
+            comments.add(commentInstance);
+        }
+
+
         Gson gson = new Gson();
         String json = gson.toJson(comments);
-
+ 
         // Send the JSON as the response
```
Comment (octopus1000) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: A general comment about "commenting", usually we use a sentence end with period.

//  Send the JSON as response.

File: portfolio/src/main/webapp/comments-script.js (hunk 4 above)
Comment (MohamedShatry) [reply portfolio/src/main/webapp/comments-script.js]: No particular date format needs to be followed. I've tried to implement toISOString(), however, it seems to be returning time in GMT. Working on fixing it

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 6 above)
Comment (MohamedShatry) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Did not know this. Will implement

File: portfolio/src/main/webapp/comments-script.js (hunk 4 above)
Comment (MohamedShatry) [reply portfolio/src/main/webapp/comments-script.js]: Figured a solution without converting it to ISOString

**Comment by MohamedShatry** [comment]
This branch has several commits. Most of these might be duplicate because I ran into some trouble with rebasing my client server into this one. 
Some changes implemented on the client side include:
-Adding an input to limit the number of comments to be received. 
-Updated the way I handled dates in javascript
-Added delete button functionality
Updated fetch requests to include parameters

Live version ca be found here : https://mshatry-step-2020.uc.r.appspot.com/

**Comment by MohamedShatry** [comment]
Merge master branch into client branch. 
Update the client branch to call new delete route

File: portfolio/src/main/webapp/comments-script.js [hunk 7]
```diff
@@ -1,7 +1,138 @@
...
+    error_tag.classList.add("error-tag");
+    const err_text = document.createTextNode("Ooops! Looks like there are no comments to show! Type one");
+    error_tag.appendChild(err_text);
+
+    main.appendChild(error);
+    main.appendChild(error_tag);
+}
+
+
+//Add an event listener to the submit button that send the response as JSON to the server.
```
Comment (octopus1000) [inline portfolio/src/main/webapp/comments-script.js:95]: Do you need to use event listener? Can we just to submit=func() on form element?

**Review by octopus1000 — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/comments-script.js (hunk 7 above)
Comment (MohamedShatry) [reply portfolio/src/main/webapp/comments-script.js:95]: I can do that. Let me update

**Comment by MohamedShatry** [comment]
Changed the event listener to a function. Also updated delete path method

**Review by octopus1000 — APPROVED** [review]

**Pull request merged**